### PR TITLE
Migrate add/remove user from channel actions to redux

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Migrate the actions to add/remove users to redux actions

Affected Areas:
* Add users to a channel using the channel menu action
* Remove users from a channel in manage channel members

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12694

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: tests/components/channel_members_dropdown.test.jsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import ChannelMembersDropdown from 'components/channel_members_dropdown/channel_members_dropdown.jsx';
+
+describe('components/channel_invite_modal', () => {
```
Comment (saturninoabril) [inline tests/components/channel_members_dropdown.test.jsx]: May be a typo error.  Should be: `components/channel_members_dropdown`

File: tests/components/channel_members_dropdown.test.jsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                };
+
+                return Promise.resolve({error});
+            }),
+            getChannelStats: jest.fn(),
+            updateChannelMemberSchemeRoles: jest.fn(),
+        },
+    };
+
+    test('should match snapshot for channel_invite_modal', () => {
```
Comment (saturninoabril) [inline tests/components/channel_members_dropdown.test.jsx]: Maybe `'should match snapshot for channel member remove button'`?

File: tests/components/channel_members_dropdown.test.jsx (hunk 1 above)
Comment (enahum) [reply tests/components/channel_members_dropdown.test.jsx]: Nice catch

File: tests/components/channel_members_dropdown.test.jsx (hunk 2 above)
Comment (enahum) [reply tests/components/channel_members_dropdown.test.jsx]: Will change

**Comment by enahum** [comment]
@saturninoabril changes made

File: actions/views/channel.js [hunk 3]
```diff
@@ -12,3 +14,15 @@ export function checkAndSetMobileView() {
         });
     };
 }
+
+export function addUsersToChannel(channelId, userIds) {
```
Comment (jwilander) [inline actions/views/channel.js]: This really isn't a view action is it? Not sure it belongs in the views module/folder

Comment (enahum) [reply actions/views/channel.js]: @jwilander why wouldn't this be a view action? actually I do think everything handle by the views should be a view action.

I mean I can make the change no problem, but I just want to understand the reasoning.

**Comment by DHaussermann** [comment]
@enahum There is a small issue with this one. When you first create a channel (that contains only you, the creator), the cont on the channel members icon will show as `-` instead of one. As soon as you leave and return to the channel it will display as 1 correctly
 
chmembercount

I was only able to repo the bug on this SpinMint. Please let me know if it is not introduced by this code change.

**Comment by enahum** [comment]
@DHaussermann this PR is only for adding and removing users to/from a channel, it does not touches anything about channel creation.

There was a PR merged to master early on that wen you create a channel everything looks about right but the address bar is from the previous channel, perhaps what you seeing ia related to that to

File: actions/views/channel.js (hunk 3 above)
Comment (jwilander) [reply actions/views/channel.js]: A view action is something local only that will only affect the view state of the client. This is adding users to a channel which is affecting the backend state and really doesn't have anything directly to do with view state.

Comment (enahum) [reply actions/views/channel.js]: alright I'll move it

**Comment by enahum** [comment]
@jwilander changes made

**Comment by DHaussermann** [comment]
Thanks @enahum . At this point I should just file a bug against master for the `-` issue. 

I'll figure out why it did not repo on bladekick.

**Comment by enahum** [comment]
@DHaussermann I have it on my radar and I’ll include a fix eith another PR that I need to submit soon, if you want you can assign that ticket to me and put it in my current sprint ;)

**Comment by DHaussermann** [comment]
@enahum No issues found related to this PR. 

Looks good to me.

**Pull request merged**